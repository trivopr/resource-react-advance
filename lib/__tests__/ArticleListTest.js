import React from 'react';
import ArticleList from '../components/ArticleList';

import renderer from 'react-test-renderer';

describe('Article List', () => {
    it('check abc', () => {
        expect('abc').toBe('abc');
    });

    const testProps = {
        articles: {
            a: {id: 'a', title: 'hello world 1'},
            b: {id: 'b', title: 'hello world 2'},
            c: {id: 'c', title: 'hello world 3'}
        },
        store: {
            lookupAuthor: jest.fn(() => ({}))
        }
    }

    it('Test Props', () => {
        const tree = renderer.create(
            <ArticleList {...testProps} />
        ).toJSON();
        console.log('TREE:', tree);
        expect(tree.children.length).toBe(3);
        expect(tree).toMatchSnapshot();
    })
})