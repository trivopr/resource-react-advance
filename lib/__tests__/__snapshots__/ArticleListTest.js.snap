// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article List Test Props 1`] = `
<div>
  <div>
    <div>
      hello world 1
    </div>
    <div>
      Invalid Date
    </div>
    <div>
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div />
    <hr />
  </div>
  <div>
    <div>
      hello world 2
    </div>
    <div>
      Invalid Date
    </div>
    <div>
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div />
    <hr />
  </div>
  <div>
    <div>
      hello world 3
    </div>
    <div>
      Invalid Date
    </div>
    <div>
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div />
    <hr />
  </div>
</div>
`;
